ledger/participant-integration-api/src/main/scala/platform/store/backend/localstore/PartyRecordStorageBackendImpl.scala

```diff
@@ -105,14 +105,30 @@ object PartyRecordStorageBackendImpl extends PartyRecordStorageBackend {
     )
   }
 
-  override def updateIdentityProviderId(
-      internalId: Int,
+  override def fetchPartiesExist(
+      parties: Set[Ref.Party],
       identityProviderId: Option[IdentityProviderId.Id],
-  )(connection: Connection): Boolean =
-    IdentityProviderAwareBackend.updateIdentityProviderId("participant_party_records")(
-      internalId,
-      identityProviderId,
-    )(
-      connection
-    )
+  )(connection: Connection): Set[Ref.Party] = {
+    import com.daml.platform.store.backend.common.SimpleSqlAsVectorOf._
+    import com.daml.platform.store.backend.common.ComposableQuery.SqlStringInterpolation
+    val filteredParties = if (parties.nonEmpty) {
+      cSQL"party in (${parties.map(_.toString)})"
+    } else {
+      cSQL"1 != 1"
+    }
+    val filteredIdentityProviderId = identityProviderId match {
+      case Some(id) => cSQL"identity_provider_id = ${id.value: String}"
+      case None => cSQL"identity_provider_id is NULL"
+    }
+    SQL"""
+         SELECT
+             party
+         FROM participant_party_records
+         WHERE
+             $filteredIdentityProviderId AND $filteredParties
+         ORDER BY party
```

Comment:
Sorting is not needed here as you're converting to a set anyway later in this method